demos/speech_recognition_offline_demo/python/speech_recognition_offline_demo.py

```diff
@@ -0,0 +1,113 @@
+        if len(decoder_output_shape) != 3:
+            raise RuntimeError('QuartzNet decoder output must be 3-dimensional')
+        if decoder_output_shape[2] != len(self.alphabet) + 1:  # +1 for blank char
+            raise RuntimeError('QuartzNet decoder second dimension must be similar to a length of an alphabet assumed')
+        net.reshape({next(iter(net.input_info)): input_shape})
+        self.exec_net = self.ie.load_network(net, device)
+
+    def infer(self, melspectrogram):
+        return next(iter(self.exec_net.infer({next(iter(self.exec_net.input_info)): melspectrogram}).values()))
+
+    @classmethod
+    def audio_to_melspectrum(cls, audio, sampling_rate):
+        preemph = 0.97
+        preemphased = np.concatenate([audio[:1], audio[1:] - preemph * audio[:-1].astype(np.float32)])
+
+        win_length = round(sampling_rate * 0.02)
+        spec = np.abs(librosa.core.spectrum.stft(preemphased, n_fft=512, hop_length=round(sampling_rate * 0.01),
+            win_length=win_length, center=True, window=scipy.signal.windows.hann(win_length), pad_mode='reflect'))
+        mel_basis = librosa.filters.mel(sampling_rate, 512, n_mels=64, fmin=0.0, fmax=8000.0, norm='slaney', htk=False)
+        log_melspectrum = np.log(np.dot(mel_basis, np.power(spec, 2)) + 2 ** -24)
+
+        normalized = (log_melspectrum - log_melspectrum.mean(1)[:, None]) / (log_melspectrum.std(1)[:, None] + 1e-5)
+        remainder = normalized.shape[1] % cls.pad_to
+        if remainder != 0:
+            return np.pad(normalized, ((0, 0), (0, cls.pad_to - remainder)))[None]
+        return normalized[None]
+
+    @classmethod
+    def ctc_greedy_decode(cls, pred):
+        prev_id = blank_id = len(cls.alphabet)
+        transcription = []
+        for idx in pred[0].argmax(1):
+            if prev_id != idx != blank_id:
+                transcription.append(cls.alphabet[idx])
+            prev_id = idx
+        return ''.join(transcription)
+
+
+def build_argparser():
+    parser = ArgumentParser(add_help=False)
+    parser.add_argument('-h', '--help', action='help', default=SUPPRESS, help='Show this help message and exit.')
+    parser.add_argument('-m', '--model', help='Required. Path to an .xml file with a trained model.', required=True)
+    parser.add_argument('-i', '--input', help="Path to an audio file in WAV PCM 16 kHz mono format", required=True)
+    parser.add_argument('-d', '--device', default='CPU',
+                        help="Optional. Specify the target device to infer on, for example: CPU, GPU, FPGA, HDDL, MYRIAD or HETERO. "
+                             "The sample will look for a suitable IE plugin for this device. Default value is CPU.")
+    return parser
+
+
+def main():
+    args = build_argparser().parse_args()
+
+    with wave.open(args.input, 'rb') as wave_read:
+        channel_num, sample_width, sampling_rate, pcm_length, compression_type, _ = wave_read.getparams()
```

Comment:
The demo description claims that only 16 KHz audio is supported, but there's no check on the sample rate here.